lib/IgnorePlugin.js

```diff
@@ -8,14 +8,20 @@
 
 class IgnorePlugin {
 	/**
-	 * @param {RegExp} resourceRegExp A RegExp to test the request against
-	 * @param {RegExp=} contextRegExp A RegExp to test the context (directory) against
+	 * @param {object} options IgnorePlugin options
+	 * @param {RegExp} options.resourceRegExp - A RegExp to test the request against
+	 * @param {RegExp} options.contextRegExp - A RegExp to test the context (directory) against
+	 * @param {TODO} options.checkResource - A function that will take resource as argument and should return true or false
+	 * @param {TODO} options.checkContext - A function that will take context as argument and should return true or false
 	 */
-	constructor(resourceRegExp, contextRegExp) {
-		/** @private @type {RegExp} */
-		this.resourceRegExp = resourceRegExp;
-		/** @private @type {RegExp} */
-		this.contextRegExp = contextRegExp;
+	constructor(options) {
+		this.options = options;
```

Comment:
provide a backward-compat interface:

``` js
// TODO webpack 5 remove this compat-layer
if(arguments.length > 1 || options instanceof RegExp) {
  options = {
    resourceRegExp: arguments[0],
    contextRegExp: arguments[1]
  };
}
```

You can send a separate PR to the `next` branch removing it again.

Reply:
I haven't worked with types in JS so far. If I do this, it complains

```
[ts]
Property 'checkResource' does not exist on type '{ resourceRegExp: RegExp; contextRegExp: RegExp; checkResource: (arg0: string) => boolean; checkC...'.
  Property 'checkResource' does not exist on type '{ [x: string]: any; resourceRegExp: any; contextRegExp: any; }'.
```

Reply:
The params need to be optional

`@param {(function(string): boolean)=} options.checkResource`